convert2rhel/subscription.py

```diff
@@ -304,29 +304,36 @@ def attach_subscription():
     pool = ["subscription-manager", "attach"]
     if tool_opts.auto_attach:
         pool.append("--auto")
-        tool_opts.pool = "-a"
```

Comment:
I don't see any comment in this PR mentioning this change. But I agree with it. The `tool_opts.pool` is not used anywhere later on. It is a leftover after removing this functionality: https://github.com/oamg/convert2rhel/pull/240.

Reply:
Aha, I see it mentioned in a message of the commit I'm going to squash soon: https://github.com/oamg/convert2rhel/pull/385/commits/7312775d3734e9ebd5c0d021d1edb4ce442c23fd

> Previous to this change, if a pool had been specified on the
convert2rhel command line and attach_subscription() was called more than
once, the subscription-manager command line would be constructed
incorrectly.  It would have "--pool" added to the front of the pool id
each time attach-subscription was called leading to something like this:
"subscription-manager attach --pool --pool POOLID".

> To fix this, we now treat tool_opts.pool as read-only and do not modify
it on each call to attach_subscription().  This goes along with our
overall move to treat global singletons as read-only whenever possible.